revengine/settings/deploy.py

```diff
@@ -10,29 +9,29 @@
 
 #### Critical settings
 
-SECRET_KEY = os.getenv("DJANGO_SECRET_KEY")
+SECRET_KEY = env.str("DJANGO_SECRET_KEY", "")
 
 ### Environment-specific settings
-ALLOWED_HOSTS = os.getenv("ALLOWED_HOSTS", "localhost").split(":")
+ALLOWED_HOSTS = env.list("ALLOWED_HOSTS", "localhost", delimeter=":")
```

Comment:
Typo here

```suggestion
ALLOWED_HOSTS = env.list("ALLOWED_HOSTS", "localhost", delimiter=":")
```

I understand why we don't unit test settings files, but this is an example of where it would be helpful to do so because catch something like this, which would potentially lead to prod being broken post deploy.  To be clear, not suggesting we pivot to testing settings.

Reply:
This is the second bug due to not having unittests for settings. https://github.com/newsrevenuehub/rev-engine/pull/1530/files#r1636745783

The pre-flight.py looks good cause default is only one-element list. If prod had longer list, and deployer was eagle-eyed, might be caught.

There's probably a reason, but really annoyed they accept all kwargs (which is why my silly import test did not fail)
```
def _preprocess_list(value, *, delimiter= ",", **kwargs):
   kwargs not used
```

I was about to write tests (for 2nd time). But, there's really so little to test, that isn't just verifying environs works. This is all I came up with 4f1bb8b05

Adding/maintaining tests for each env.list() and similar; seems not ROI worthy. These are issues where we have to rely on reviews and QA testing to catch. IMHO.